src/main/java/org/gingesnap/k8s/KubernetesClientProducer.java

```diff
@@ -0,0 +1,41 @@
+package org.gingesnap.k8s;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.quarkus.arc.lookup.LookupIfProperty;
+
+@Singleton
+public class KubernetesClientProducer {
+
+   private static final Logger log = LoggerFactory.getLogger(KubernetesClientProducer.class);
+
+   @Produces
+   @LookupIfProperty(name = "gingernsnap.k8s.enabled", stringValue = "true")
+   public KubernetesClient kubernetesClient() {
+      log.info("Creating Kubernetes client");
+
+      try {
+         // TODO hardcoded
+         String namespace = Files.readString(Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace"));
```

Comment:
I know this approach is in the cache-manager code, but it's probably a better approach to make it so that the namespace is a mandatory configuration option. This will make it easier to execute the code locally, as the KubernetesClient will just use the local KUBE_CONFIG  context and startup won't fail due to ^ file not existing on the local machine.